torch/_dynamo/variables/constant.py

```diff
@@ -98,7 +108,19 @@ def has_arith_binop(num_ty):
             return ConstantVariable(method(*const_args, **const_kwargs), **options)
         elif has_arith_binop(int) or has_arith_binop(float):
             op = getattr(operator, name)
-            return ConstantVariable(op(self.value, const_args[0]), **options)
+            add_target = const_args[0]
+            if isinstance(add_target, (torch.SymInt, torch.SymFloat)):
+                from .tensor import DynamicShapeVariable
+
+                # Addition between a non sym and sym makes a sym
+                dyn_shape = tx.output.register_attr_or_module(
```

Comment:
Why does it get registered here 🤔

Reply:
So we can be aware of it. Anytime you create a new tensor/module/symint/foo in dynamo that is referenced by proxy, you need to register it somewhere.

Reply:
It still seems crazy pants to encounter an untracked symbolic integer, and decide that the "oh, I guess the way I'll model this is by putting an integer(!!) attribute on the produced nn module".

Like, how did the untracked symbolic integer even get produced in the first place, the only way we're manufacturing these is by setting symbolic sizes/strides on tensors lol

Reply:
It's a little crazy pants. I guess, sorry, to frame it differently, I am *making* a new integer here. So that's how it got produced. I am saying we found a case where we are doing:

Constant + SymInt0

So, what goes into the output?

I posit that it is:

Constant + SymInt0 = SymInt1

So, we need to register SymInt1 somewhere OR learn to store functions on constants in the graph. As the latter is overcomplicated and not supported, we just do the operation here and let SymInt1 float through the rest of the program. However, in doing so, we need to make a proxy so that if SymInt1 becomes an arg input to something else, or an output, we can actually access it again. Hence, we need to register it somewhere. This is that. I am super open to better ways of doing this.

Reply:
We can store functions on constants in the graph... in fact, make_fx goes through considerable lengths to represent functions on constants by directly putting the constants in the args list (and not first wrapping them in a SymInt and then adding them)

Reply:
Oh interesting, I recall doing this because we had an assert - but maybe that was just a coverage thing I papered over with this. Let me play with on the symbolic-shape branch and then get back to this.

Reply:
Fixed to match the patterns here better.

Reply:
not pushed yet, I guess